src/test/java/ca/on/oicr/gsi/dimsum/MockCase.java

```diff
@@ -412,16 +416,43 @@ private static Case makeCase32() {
   private static Case makeCase33() {
     // Full-depth completed, analysis review skipped
     final int caseNumber = 33;
-    Case kase = makeCase("PRO33_0001", "Single Test", "PRO33", "REQ33", caseNumber);
+    Case kase = makeCase("PRO33_0001", "Single Test", "PRO33", "REQ33", caseNumber, null, null);
     addTest(kase, caseNumber, 1, "test", "WG", true, true, true, true);
     addDeliverable(kase, DeliverableType.DATA_RELEASE, "FastQ");
     Project project = kase.getProjects().iterator().next();
     when(project.isAnalysisReviewSkipped()).thenReturn(true);
     return kase;
   }
 
+  private static Case makeCase34() {
+    final int caseNumber = 34;
+    LocalDate startDate = LocalDate.of(2024, 1, 1);
+    LocalDate completionDate = LocalDate.of(2024, 1, 15);
+    Case kase =
+        makeCase("PRO34_0001", "Single Test", "PRO34", "REQ34", caseNumber, startDate,
+            completionDate);
+
+    // Mocking the completion logic based on deliverable states
+    List<CaseDeliverable> deliverables = new ArrayList<>();
+    CaseDeliverable deliverable = mock(CaseDeliverable.class);
+    when(deliverable.getReleaseApprovalQcPassed()).thenReturn(true);
+    when(deliverable.getAnalysisReviewQcPassed()).thenReturn(true);
+    CaseRelease release = mock(CaseRelease.class);
+    when(release.getQcPassed()).thenReturn(true);
+    when(release.getQcDate()).thenReturn(completionDate);
+    when(deliverable.getReleases()).thenReturn(Arrays.asList(release));
+    deliverables.add(deliverable);
+
+    when(kase.getDeliverables()).thenReturn(deliverables);
+
+    addCompletionDate(kase, completionDate);
+
+    return kase;
+  }
+
   private static Case makeCase(String donorName, String assayName, String projectName,
-      String requisitionName, int caseNumber) {
+      String requisitionName, int caseNumber, LocalDate startDate,
+      LocalDate completionDate) {
```

Comment:
These should be made to function like normal cases in any way that matters to the tests. Start date should never be null